### PR TITLE
Fix federationStrategy 'allowAll'

### PR DESCRIPTION
**Pull request by akshaymankar**
https://wearezeta.atlassian.net/browse/WPB-3796

Supersedes #3526

## Checklist

 - [x] Add a new entry in an appropriate subdirectory of `changelog.d`
 - [x] Read and follow the [PR guidelines](https://docs.wire.com/developer/developer/pr-guidelines.html)

### CODE REVIEW

File: integration/test/Test/Brig.hs [hunk 1]
```diff
@@ -1,11 +1,12 @@
+{-# OPTIONS_GHC -Wno-incomplete-uni-patterns #-}
```
Comment (fisx) [inline integration/test/Test/Brig.hs:1]: i'm also ok with doing this in integration.cabal...

File: services/brig/test/integration/Federation/End2end.hs [hunk 2]
```diff
@@ -114,22 +112,6 @@ spec _brigOpts mg brig galley cargohold cannon _federator brigTwo galleyTwo carg
...
-  resultViaBrigTwo <- getUserInfoFromHandle brigTwo domain handle
-
-  -- query the local-namespace brig for a user sitting on the other backend
-  -- (which will exercise the network traffic via two federators to the remote brig)
-  resultViaBrigOne <- getUserInfoFromHandle brig domain handle
-
-  liftIO $ assertEqual "remote handle lookup via federator should work in the happy case" (profileQualifiedId resultViaBrigOne) (userQualifiedId userBrigTwo)
-  liftIO $ assertEqual "querying brig1 or brig2 about the same user should give same result" resultViaBrigTwo resultViaBrigOne
-
 testSearchUsers :: Brig -> Brig -> Http ()
```
Comment (fisx) [inline services/brig/test/integration/Federation/End2end.hs:115]: this isn't called anywhere, but neither is it translated into `/integration`.  i think running this won't pass for the same reason `testHandleLookup` fails?

File: integration/test/Test/Demo.hs [hunk 3]
```diff
@@ -174,22 +175,10 @@ testIndependentESIndices = do
...
-    [aDynDomain, anotherDynDomain] <- pure dynDomains
-    _ <- Internal.createFedConn anotherDynDomain (Internal.FedConn aDynDomain "full_search")
-    threadDelay 2_000_000
-
-    u1 <- randomUser aDynDomain def
-    u2 <- randomUser anotherDynDomain def
-    uid2 <- objId u2
-    Internal.refreshIndex anotherDynDomain
-
-    bindResponse (Public.searchContacts u1 (u2 %. "name") anotherDynDomain) $ \resp -> do
```
Comment (fisx) [inline integration/test/Test/Demo.hs:187]: did you remove this search intentionally?

**Review by fisx — APPROVED** [review]

File: services/brig/test/integration/Federation/End2end.hs (hunk 2 above)
Comment (akshaymankar) [reply services/brig/test/integration/Federation/End2end.hs]: Deleted it now.
The test already existed in the new test suite: https://github.com/wireapp/wire-server/blob/1c6f7463d9eacfe68cdc402952e41213869267c0/integration/test/Test/Brig.hs#L144-L160

File: integration/test/Test/Demo.hs (hunk 3 above)
Comment (akshaymankar) [reply integration/test/Test/Demo.hs:187]: Yes, the test is supposed to show that federation works. Using search here is just making the test more complicated.

File: integration/test/Test/Brig.hs (hunk 1 above)
Comment (akshaymankar) [reply integration/test/Test/Brig.hs:1]: Create a PR then.

**Pull request merged**